src/main/scala/is/hail/rvd/RVDContext.scala

```diff
@@ -0,0 +1,27 @@
+package is.hail.rvd
+
+import is.hail.annotations.Region
+
+object RVDContext {
+  def default: RVDContext = SimpleRVDContext(Region())
+
+  def fromRegion(region: Region): RVDContext = SimpleRVDContext(region)
+}
+
+// NB: must be *Auto*Closeable because calling close twice is undefined behavior
+// (see AutoCloseable javadoc)
+trait RVDContext extends AutoCloseable {
+  def region: Region // lifetime: element
+
+  def partitionRegion: Region // lifetime: partition
+
+  // frees the memory associated with this context
+  def close: Unit
```

Comment:
On AutoCloseable: you also have the option of making Region.free idempotent by nulling out the pointed-to storage (a good idea in general) and checking for null before freeing:

```
  if (mem != null) {
    Memory.free(mem)
    mem = null
  }
```